backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/validator/HostValidator.java

```diff
@@ -162,7 +162,8 @@ protected VDS getHost() {
 
     public ValidationResult validateStatusForEnrollCertificate() {
         return ValidationResult.failWith(EngineMessage.CANNOT_ENROLL_CERTIFICATE_HOST_STATUS_ILLEGAL)
-                .unless(host.getStatus() == VDSStatus.Maintenance || host.getStatus() == VDSStatus.InstallFailed);
+                .unless(host.getStatus() == VDSStatus.Maintenance || host.getStatus() == VDSStatus.InstallFailed
+                        || host.getStatus() == VDSStatus.NonResponsive);
```

Comment:
I guess we have no way to distinguish between hosts that we don't hear from due to invalid certification and for other reasons, right?

Reply:
right. unfortunately

Reply:
It seems we currently don't track the reasons behind VDSStatus.NonResponsive nor attempt to check for anything more sophisticated in the stuff around VDSNetworkException.  I'm afraid it wouldn't be easy to add reliable checks and tracking.

The risk here may be that users may attempt to renew the certificates "just in case" if the host is non-responsive.  If it succeeds it will renew the certificates, which should be harmless, and restart the services, which may not be that innocent but probably also not that bad considering the host is in non-responsive state while still being reachable.

Reply:
yeah, and issuing certificate renewal automatically every time the host becomes non-responsive doesn't sound that good either..